addon-manager/impl/src/main/java/org/jboss/forge/addon/manager/impl/ui/DistributionUpdateCommand.java

```diff
@@ -0,0 +1,140 @@
+   public Result updateDistribution(UIExecutionContext context) throws Exception
+   {
+      String forgeHomeDirPath = OperatingSystemUtils.getForgeHomeDir().getAbsolutePath();
+      
+      if(updateDir.exists()) {
+         return Results.fail("There is already pending update in this forge distribution."
+                  + " Restart forge to finish the update or "
+                  + "run update-distribution-cancel command to cancel the updating process");
+      }
+      if (!context.getPrompt().promptBoolean(
+               "Are you sure you want to continue? This command will delete current directories: addons, bin, lib"))
+      {
+         return Results.fail("Updating distribution was aborted.");
+      }
+
+      List<Coordinate> distributionVersions = dependencyResolver.resolveVersions(DependencyQueryBuilder
+               .create(CoordinateBuilder.create()
+                        .setGroupId("org.jboss.forge")
+                        .setArtifactId("forge-distribution")));
+      Coordinate latestDistributionCoordinate;
+      int i = 0;
+      do
+      {
+         i++;
+         if (distributionVersions.size() <= i)
+         {
+            return Results.fail("Have not found any online Final version of the distribution.");
+         }
+         latestDistributionCoordinate = distributionVersions.get(distributionVersions.size() - i);
+      }
+      while (latestDistributionCoordinate.isSnapshot());
+      
+      Version latestVersion = new SingleVersion(latestDistributionCoordinate.getVersion());
+      Version installedVersion = currentAddon.getId().getVersion();
+      if (latestVersion.compareTo(installedVersion) <= 0)
+      {
+         return Results.fail("Forge is already in the most updated version.");
+      }
+
+      Dependency dependency = dependencyResolver.resolveArtifact(DependencyQueryBuilder.create(CoordinateBuilder
+               .create()
+               .setGroupId(latestDistributionCoordinate.getGroupId())
+               .setArtifactId(latestDistributionCoordinate.getArtifactId())
+               .setVersion(latestDistributionCoordinate.getVersion()).setClassifier("offline").setPackaging("zip")));
+      FileResource<?> artifact = dependency.getArtifact();
+      ZipFile zipFile = new ZipFile(artifact.getFullyQualifiedName());
+      FileHeader forgeDistDir = (FileHeader)zipFile.getFileHeaders().get(0);
+      zipFile.extractAll(forgeHomeDirPath);
+      File extractedNewDis = new File(forgeHomeDir,forgeDistDir.getFileName());
+      extractedNewDis.renameTo(updateDir);
+      return Results.success("Sucessfully prepared to be updated to version "
+               + latestDistributionCoordinate.getVersion()
+               + ". Please restart forge.");
+   }
+   
+   @Command(value="Forge: Cancel Update Distribution",categories={ "Forge", "Manage" }, help="Cancel updating the distribution", enabled=NonGUIEnabledPredicate.class)
```

Comment:
It would be better if this command was enabled only if the .update directory existed